src/codegen/sdk/codebase/codebase_context.py

```diff
@@ -319,10 +320,28 @@ def build_directory_tree(self, files: list[SourceFile]) -> None:
         """Builds the directory tree for the codebase"""
         # Reset and rebuild the directory tree
         self.directories = dict()
+        created_dirs = set()
         for file in files:
             directory = self.get_directory(file.path.parent, create_on_missing=True)
             directory.add_file(file)
             file._set_directory(directory)
+            created_dirs.add(file.path.parent)
+
+        def _dir_has_file(filepath):
+            gen = os.scandir(filepath)
+            while entry := next(gen, None):
+                if entry.is_file():
+                    return True
+            return False
+
+        for rel_filepath in self._op.get_filepaths_for_repo(GLOBAL_FILE_IGNORE_LIST):
```

Comment:
Can we actually iterate over each context's repo operator instead of setting self._op?